File: ValidPalindarome/isPalindrome.java
public class Solution {
    public boolean isPalindrome(String s) {
        if(s.isEmpty()) return true;
        int head = 0, tail = s.length() - 1;
        char chead,ctail;
        while( head<= tail) {
            chead = s.charAt(head);
            ctail = s.charAt(tail);
            if(!Character.isLetterOrDigit(chead)) {
                head ++;
            } else if(!Character.isLetterOrDigit(ctail)) {
                tail --;
            } else {
                if(Character.toLowerCase(chead) != Character.toLowerCase(ctail)) return false;
                else {head++;tail--;}
            }
        }
        return true;
    }
}

File: src/ValidPalindarome/isPalindrome.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isPalindrome = function(s) {
    if(s == '') return true;
    var s = s.toLowerCase().replace(/[^a-z0-9]/g,'');
    for(var i = 0; i <= s.length/2; i++) {
        if(s.charAt(i) != s.charAt(s.length - 1 - i)) return false;
    }
    return true;
};